routes/baseActionsRouter.js

```diff
@@ -496,6 +496,27 @@ router.get("/decoder", function(req, res, next) {
 	next();
 });
 
+allSettled = function(promiseList) {
+    let results = new Array(promiseList.length);
+
+    return new Promise((ok, rej) => {
```

Comment:
why `rej` is never used?

Reply:
The ok, rej callback model of promises does not work when some of the promises might have succeeded and others been rejected.  Therefore, allSettled always calls "ok" when every promise is settled with a list containing each promise's result.  Its up to the handler to determine whether the promise was accepted or rejected.  Therefore the "rej" callback is a no-op -- its an unused leftover needed because promises need a function with 2 args (actually, given how "loose" javascript is with args, I'm not actually sure if you have to keep to the standard Promise structure, but I did it to be safe).